templates/default/redhat/init.d/chef-client.erb

```diff
@@ -43,6 +43,8 @@ start() {
 
 stop() {
     echo -n $"Stopping $prog: "
+    # It will pick the process id directly from the running process and adding into $pidfile.
+    ps -ef |grep $exec |grep -v grep |awk '{print $2}' > $pidfile
```

Comment:
Has this been tested to confirm this works. -p is for a pidfile. Does it work if you stick a PID in there instead of a path?

Reply:
@tas50 Yes it has been testes and working fine. It won't work if we stick PID in there because `killproc` does not having that type of functionality. In `killproc` function there are mentioned that if you are mentioning -p then you have to give pidfile not pid.